File: tests/adapters/python/goldens/do/literals.py
"""Module for testing literal data optimization."""

import json

# Short string literal (should be preserved)
SHORT_STRING = "hello"

# Long string literal (candidate for trimming)
LONG_STRING = """This is a very long string that contains a lot of text and might be a candidate for trimming when optimizing code for AI context windows. It has multiple sentences and provides detailed explanations that might not be essential for understanding the code structure. This string continues with even more verbose content to ensure it exceeds typical trimming thresholds."""

# Multi-line string with data
DATA_STRING = """
{
    "users": [
        {"id": 1, "name": "Alice", "email": "alice@example.com"},
        {"id": 2, "name": "Bob", "email": "bob@example.com"},
        {"id": 3, "name": "Charlie", "email": "charlie@example.com"}
    ],
    "metadata": {
        "version": "1.0",
        "timestamp": "2024-01-15T10:30:00Z",
        "description": "Sample user data for testing purposes"
    }
}
"""

class DataContainer:
    """Class with various literal types."""
    
    def __init__(self):
        # Small array (should be preserved)
        self.small_list = [1, 2, 3]
        
        # Large array (candidate for trimming)
        self.large_list = [
            "item_1", "item_2", "item_3", "item_4", "item_5",
            "item_6", "item_7", "item_8", "item_9", "item_10",
            "item_11", "item_12", "item_13", "item_14", "item_15",
            "item_16", "item_17", "item_18", "item_19", "item_20",
            "item_21", "item_22", "item_23", "item_24", "item_25"
        ]
        
        # Small dictionary (should be preserved)
        self.small_dict = {"name": "test", "value": 42}
        
        # Large dictionary (candidate for trimming)
        self.large_dict = {
            "user_id": 12345,
            "username": "john_doe",
            "email": "john.doe@example.com",
            "first_name": "John",
            "last_name": "Doe",
            "age": 30,
            "address": {
                "street": "123 Main St",
                "city": "Anytown",
                "state": "CA",
                "zip_code": "12345",
                "country": "USA"
            },
            "preferences": {
                "theme": "dark",
                "language": "en",
                "notifications": True,
                "newsletter": False
            },
            "roles": ["user", "beta_tester", "premium"],
            "last_login": "2024-01-15T10:30:00Z",
            "created_at": "2023-01-01T00:00:00Z",
            "updated_at": "2024-01-15T10:30:00Z",
            "is_active": True,
            "settings": {
                "privacy": "public",
                "two_factor": True,
                "session_timeout": 3600
            }
        }

def process_data():
    """Function with various literal data."""
    # Multi-line list
    categories = [
        "Technology", "Science", "Health", "Education", "Entertainment",
        "Sports", "Travel", "Food", "Fashion", "Business", "Politics",
        "Environment", "Art", "Music", "Literature", "History"
    ]
    
    # Nested data structure
    config = {
        "database": {
            "host": "localhost",
            "port": 5432,
            "name": "myapp",
            "credentials": {
                "username": "admin",
                "password": "super_secret_password_123456789"
            }
        },
        "api": {
            "endpoints": [
                "/api/v1/users",
                "/api/v1/posts", 
                "/api/v1/comments",
                "/api/v1/categories",
                "/api/v1/tags"
            ],
            "rate_limit": 1000,
            "timeout": 30
        },
        "features": {
            "authentication": True,
            "caching": True,
            "logging": True,
            "monitoring": True,
            "compression": False
        }
    }
    
    # Very long single-line string
    sql_query = "SELECT users.id, users.username, users.email, profiles.first_name, profiles.last_name, profiles.bio, addresses.street, addresses.city, addresses.state, addresses.zip_code FROM users JOIN profiles ON users.id = profiles.user_id JOIN addresses ON users.id = addresses.user_id WHERE users.is_active = true AND profiles.is_public = true ORDER BY users.created_at DESC LIMIT 100"
    
    return categories, config, sql_query

# Set literal
TAGS = {
    "python", "javascript", "typescript", "java", "csharp", "cpp", "rust",
    "go", "kotlin", "swift", "php", "ruby", "scala", "clojure", "haskell"
}

# Tuple with many elements
COORDINATES = (
    (0, 0), (1, 1), (2, 2), (3, 3), (4, 4), (5, 5),
    (6, 6), (7, 7), (8, 8), (9, 9), (10, 10), (11, 11)
)

File: tests/adapters/rust/goldens/do/literals.rs
// Rust module for testing literal optimization.

use std::collections::HashMap;

// Short string literal (should be preserved)
const SHORT_MESSAGE: &str = "Hello, World!";

// Long string literal (candidate for trimming)
const LONG_MESSAGE: &str = "This is an extremely long message that contains a substantial amount of text content which might be considered for trimming when optimizing Rust code for AI context windows. The message continues with detailed explanations and verbose descriptions that may not be essential for understanding the core functionality and structure of the code. This string literal spans multiple conceptual lines even though it's defined as a single string literal.";

// Multi-line string with formatting
const TEMPLATE_WITH_DATA: &str = r#"User Information:
- Name: {}
- Email: {}
- Registration Date: {}
- Account Status: {}
- Permissions: {:?}
- Last Login: {}
- Profile Completeness: {}%"#;

#[derive(Debug)]
struct DataContainer {
    // Small vec (should be preserved)
    tags: Vec<String>,

    // Large vec (candidate for trimming)
    items: Vec<String>,

    // Small map (should be preserved)
    metadata: HashMap<String, String>,

    // Large map (candidate for trimming)
    configuration: HashMap<String, HashMap<String, i32>>,
}

struct LiteralDataManager {
    // Small config (should be preserved)
    small_config: HashMap<String, bool>,

    // Large config (candidate for trimming)
    large_config: HashMap<String, HashMap<String, serde_json::Value>>,

    // Supported languages
    supported_languages: Vec<String>,

    // Allowed extensions
    allowed_extensions: Vec<String>,
}

impl LiteralDataManager {
    fn new() -> Self {
        let mut small_config = HashMap::new();
        small_config.insert("debug".to_string(), true);
        small_config.insert("version".to_string(), false);

        let large_config = {
            let mut config = HashMap::new();

            let mut database = HashMap::new();
            database.insert("host".to_string(), json!("localhost"));
            database.insert("port".to_string(), json!(5432));
            database.insert("name".to_string(), json!("application_db"));
            database.insert("ssl".to_string(), json!(false));
            database.insert("pool_min".to_string(), json!(2));
            database.insert("pool_max".to_string(), json!(10));
            database.insert("idle_timeout".to_string(), json!(30000));
            database.insert("connection_timeout".to_string(), json!(2000));
            database.insert("retry_attempts".to_string(), json!(3));
            database.insert("retry_delay".to_string(), json!(1000));
            config.insert("database".to_string(), database);

            let mut cache = HashMap::new();
            cache.insert("redis_port".to_string(), json!(6379));
            cache.insert("redis_db".to_string(), json!(0));
            cache.insert("redis_ttl".to_string(), json!(3600));
            cache.insert("memory_max_size".to_string(), json!(1000));
            cache.insert("memory_ttl".to_string(), json!(1800));
            config.insert("cache".to_string(), cache);

            let mut api = HashMap::new();
            api.insert("timeout".to_string(), json!(30000));
            api.insert("retries".to_string(), json!(3));
            api.insert("rate_limit_requests".to_string(), json!(100));
            api.insert("rate_limit_window".to_string(), json!(60000));
            config.insert("api".to_string(), api);

            let mut features = HashMap::new();
            features.insert("authentication".to_string(), json!(true));
            features.insert("authorization".to_string(), json!(true));
            features.insert("logging".to_string(), json!(true));
            features.insert("monitoring".to_string(), json!(true));
            features.insert("analytics".to_string(), json!(false));
            features.insert("caching".to_string(), json!(true));
            features.insert("compression".to_string(), json!(true));
            config.insert("features".to_string(), features);

            config
        };

        let supported_languages = vec![
            "english", "spanish", "french", "german", "italian", "portuguese",
            "russian", "chinese", "japanese", "korean", "arabic", "hindi",
            "dutch", "swedish", "norwegian", "danish", "finnish", "polish",
            "czech", "hungarian", "romanian", "bulgarian", "croatian", "serbian",
        ].into_iter().map(String::from).collect();

        let allowed_extensions = vec![
            ".rs",
            ".py", ".js", ".ts", ".jsx", ".tsx",
            ".java", ".kt", ".scala",
            ".cpp", ".cxx", ".cc", ".c", ".h", ".hpp", ".hxx",
            ".cs", ".go",
            ".php", ".rb", ".swift", ".clj",
        ].into_iter().map(String::from).collect();

        Self {
            small_config,
            large_config,
            supported_languages,
            allowed_extensions,
        }
    }

    fn process_data(&self) -> DataContainer {
        let small_vec = vec!["one".to_string(), "two".to_string(), "three".to_string()];

        let large_vec = vec![
            "item_001", "item_002", "item_003", "item_004", "item_005",
            "item_006", "item_007", "item_008", "item_009", "item_010",
            "item_011", "item_012", "item_013", "item_014", "item_015",
            "item_016", "item_017", "item_018", "item_019", "item_020",
            "item_021", "item_022", "item_023", "item_024", "item_025",
            "item_026", "item_027", "item_028", "item_029", "item_030",
        ].into_iter().map(String::from).collect();

        let mut metadata = HashMap::new();
        metadata.insert("type".to_string(), "test".to_string());
        metadata.insert("count".to_string(), "3".to_string());

        DataContainer {
            tags: small_vec,
            items: large_vec,
            metadata,
            configuration: HashMap::new(),
        }
    }

    fn get_long_query(&self) -> &str {
        r#"
SELECT
    users.id, users.username, users.email, users.created_at,
    profiles.first_name, profiles.last_name, profiles.bio, profiles.avatar_url,
    addresses.street, addresses.city, addresses.state, addresses.postal_code, addresses.country,
    subscriptions.plan_name, subscriptions.status, subscriptions.expires_at,
    payments.amount, payments.currency, payments.payment_date, payments.method
FROM users
LEFT JOIN profiles ON users.id = profiles.user_id
LEFT JOIN addresses ON users.id = addresses.user_id
LEFT JOIN subscriptions ON users.id = subscriptions.user_id
LEFT JOIN payments ON users.id = payments.user_id
WHERE users.is_active = true
    AND users.email_verified = true
    AND profiles.is_public = true
    AND subscriptions.status IN ('active', 'trial')
ORDER BY users.created_at DESC, subscriptions.expires_at ASC
LIMIT 100 OFFSET 0
        "#
    }
}

// Module-level constants with different sizes
struct SmallConstants;
impl SmallConstants {
    const API_VERSION: &'static str = "v1";
    const DEFAULT_LIMIT: i32 = 50;
}

fn main() {
    let manager = LiteralDataManager::new();
    let data = manager.process_data();

    println!("Tags: {:?}", data.tags);
    println!("Items count: {}", data.items.len());
    println!("{}", manager.get_long_query());
}
